### PR TITLE
ci: Add GH workflow to validate PR titles follow Conventional Commits (fixes #43).

### LINKED ISSUE #43: Automate PR title check
### Request

We should automate PR title check as we did in other repos, e.g., [log-surgeon](https://github.com/y-scope/log-surgeon/pull/51)

### Possible implementation

https://github.com/y-scope/log-surgeon/pull/51

### PR DESCRIPTION
**Pull request by LinZhihao-723**
# References

# Description

This PR implements #43 

[Convention Commits](https://www.conventionalcommits.org) is a specification for writing commit messages (or in our case, PR titles) that makes it easy to see at a glance what change the commit makes which in turn makes it easier to generate release notes.

This PR adds a workflow to check PR titles match the spec.

# Validation performed

- Already validated this workflow in other repos, including CLP core, log surgeon, and clp-ffi-py.

## Summary by CodeRabbit

- **New Features**
	- Introduced a new GitHub Actions workflow to validate pull request titles based on the Conventional Commits specification.
	- Updated the pull request template to provide guidance on writing effective titles.

- **Documentation**
	- Removed the "References" section from the pull request template and added comments to improve clarity on title formatting.

### CODE REVIEW

File: .github/workflows/pr-title-checks.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+name: "pr-title-checks"
+
+on:
+  pull_request_target:
+    types: ["edited", "opened", "reopened"]
+    branches: ["main"]
```
[the remark this answers is not in the document]
Comment (LinZhihao-723) [reply .github/workflows/pr-title-checks.yaml]: what do u think? @kirkrodrigues

[the remark this answers is not in the document]
Comment (kirkrodrigues) [reply .github/workflows/pr-title-checks.yaml]: * `pull_request_target` is [necessary](https://github.com/amannn/action-semantic-pull-request/tree/main?tab=readme-ov-file#event-triggers) for the action to work for PRs from external contributors.
* Following GitHub's [recommendation](https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#pull_request_target) in the warning, we do scope the permissions down to just `read`.
* I suppose it's still worth adding a comment in the workflow based on GitHub's warning: "you should make sure that you do not check out, build, or run untrusted code from the pull request with this event."

[the remark this answers is not in the document]
Comment (LinZhihao-723) [reply .github/workflows/pr-title-checks.yaml]: Added in the latest commit

File: .github/workflows/pr-title-checks.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+on:
+  pull_request_target:
+    # NOTE: The `pull_request_target` event means GITHUB_TOKEN can access secrets and is granted
+    # read/write repository access by default. So we need to ensure:
+    # - This workflow doesn't inadvertently check out, build, or execute untrusted code from the
+    #   pull request triggered by this event.
+    # - Each job has `permissions` set to only those necessary.
+    types: ["edited", "opened", "reopened"]
+    branches: ["main"]
+
```
Comment (kirkrodrigues) [inline .github/workflows/pr-title-checks.yaml:12]:
```suggestion
permissions: {}

```

File: .github/workflows/pr-title-checks.yaml [hunk 3]
```diff
@@ -0,0 +1,28 @@
+name: "pr-title-checks"
+
+on:
+  pull_request_target:
+    # NOTE: The `pull_request_target` event means GITHUB_TOKEN can access secrets and is granted
+    # read/write repository access by default. So we need to ensure:
```
Comment (kirkrodrigues) [inline .github/workflows/pr-title-checks.yaml]:
```suggestion
    # NOTE: Workflows triggered by this event give the workflow access to secrets and grant the
    # `GITHUB_TOKEN` read/write repository access by default. So we need to ensure:
```

**Review by kirkrodrigues — CHANGES_REQUESTED** [review]

**Review by kirkrodrigues — APPROVED** [review]

**Pull request merged**